src/main/java/org/folio/service/export/LocalFileSystemExportService.java

```diff
@@ -1,62 +1,114 @@
+  private static final String INSTANCE_ID_KEY = "i";
+  private static final String SUBFIELDS_KEY = "subfields";
+  private static final int SINGLE_INSTANCE_INDEX = 0;
+  private static final String INSTANCES = "instances";
+  private static final int SINGLE_INSTANCE = 1;
+
   @Autowired
   @Qualifier("LocalFileSystemStorage")
   private FileStorage fileStorage;
   @Autowired
   private ExportStorageService exportStorageService;
   @Autowired
   private ErrorLogService errorLogService;
+  @Autowired
+  private InventoryClient inventoryClient;
 
   @Override
-  public void exportSrsRecord(List<String> jsonRecords, FileDefinition fileDefinition) {
+  public void exportSrsRecord(List<String> jsonRecords, ExportPayload exportPayload) {
+    FileDefinition fileDefinition = exportPayload.getFileExportDefinition();
+    String jobExecutionId = exportPayload.getJobExecutionId();
+    OkapiConnectionParams params = exportPayload.getOkapiConnectionParams();
     if (CollectionUtils.isNotEmpty(jsonRecords) && fileDefinition != null) {
       for (String jsonRecord : jsonRecords) {
-        byte[] bytes = convertJsonRecordToMarcRecord(jsonRecord);
         try {
+          byte[] bytes = convertJsonRecordToMarcRecord(jsonRecord);
           if (isNotEmpty(bytes)) {
             fileStorage.saveFileDataBlocking(bytes, fileDefinition);
           }
+        } catch (MarcException e) {
+          String instId = getInstanceIdFromMarcRecord(new JsonObject(jsonRecord));
+          inventoryClient.getInstancesByIds(Collections.singletonList(instId), jobExecutionId, params, SINGLE_INSTANCE).ifPresent(instancesJson -> {
+            JsonArray instances = instancesJson.getJsonArray(INSTANCES);
+            errorLogService.saveWithAffectedRecord(instances.getJsonObject(SINGLE_INSTANCE_INDEX), ERROR_MARC_RECORD_CANNOT_BE_CONVERTED.getCode(), jobExecutionId, e, params);
+          });
         } catch (RuntimeException e) {
           LOGGER.error("Error during saving srs record to file with content: {}", jsonRecord);
         }
       }
     }
   }
 
+  @SuppressWarnings("OptionalGetWithoutIsPresent")
+  private String getInstanceIdFromMarcRecord(JsonObject marcRecord) {
+    JsonArray fields = marcRecord.getJsonArray("fields");
+    JsonObject instanceIdHolderField = fields.stream()
+      .map(JsonObject.class::cast)
+      .filter(jsonObject -> {
+        if (jsonObject.containsKey(GENERAL_INFO_FIELD_TAG_NUMBER)) {
+          JsonObject dataFieldContent = jsonObject.getJsonObject(GENERAL_INFO_FIELD_TAG_NUMBER);
+          String firstIndicator = dataFieldContent.getString(FIRST_INDICATOR);
+          String secondIndicator = dataFieldContent.getString(SECOND_INDICATOR);
+          return StringUtils.isNotBlank(firstIndicator) && StringUtils.isNotBlank(secondIndicator)
+            && firstIndicator.equals(secondIndicator) && firstIndicator.equals(INDICATOR_VALUE);
+        }
+        return false;
+      }).findFirst().get();
```

Comment:
Checking if something exist related conditions may useful here.

Reply:
@alekGbuz , I didn't put this check since it is not necessary here because every record has a 999 field with record id and instance id subfields. As well in this particular case to this method records that are passed always have related instances therefore the case when the 999 field has not instance id is not possible.